modules/packages/Python.chpl

```diff
@@ -2039,8 +2079,136 @@ module Python {
+    */
+    proc size: int throws {
+      var g = PyGILState_Ensure();
+      defer PyGILState_Release(g);
+
+      var size = PyTuple_Size(this.getPyObject());
+      this.interpreter.checkException();
+      return size;
+    }
+
+    /*
+      Get an item from the tuple. Equivalent to calling ``obj[idx]`` in Python.
+
+      :arg T: The Chapel type of the item to return.
+      :arg idx: The index of the item to get.
+      :returns: The item at the given index.
+    */
+    pragma "docs only"
+    proc get(type T = owned Value, idx: int): T throws do
+      compilerError("docs only");
+
+    @chpldoc.nodoc
+    proc get(type T, idx: int): T throws {
+      var g = PyGILState_Ensure();
+      defer PyGILState_Release(g);
+
+      var item = PyTuple_GetItem(this.getPyObject(), idx.safeCast(Py_ssize_t));
+      this.interpreter.checkException();
+      Py_INCREF(item);
+      return interpreter.fromPythonInner(T, item);
+    }
+
+    @chpldoc.nodoc
+    proc get(idx: int): owned Value throws do
+      return this.get(owned Value, idx);
+
+    /*
+      Get a slice of the tuple indicated by ``bounds``.  Equivalent to
+      calling ``obj[bounds.low:bounds.high+1]`` in Python.
+
+      .. note::
+
+         This method does not support strided ranges or ranges with an alignment
+         other than 0.
+
+      :arg T: the Chapel type of the slice to return.
+      :arg bounds: The full slice of the tuple to return
+      :returns: A slice of the tuple for the given bounds.
+    */
+    pragma "docs only"
+    proc get(type T = owned Value, bounds: range(?)): T throws do
+      compilerError("docs only");
+
+    @chpldoc.nodoc
+    proc get(type T, bounds: range(?)): T throws {
+      if (bounds.strides != strideKind.one) {
+        compilerError("cannot call `get()` on a Python tuple with a range with stride other than 1");
+      }
+
+      var g = PyGILState_Ensure();
+      defer PyGILState_Release(g);
+
+      var pyObj;
+
+      if (bounds.hasLowBound() && bounds.hasHighBound()) {
+        pyObj = PyTuple_GetSlice(this.getPyObject(),
+                                 bounds.low.safeCast(Py_ssize_t),
+                                 bounds.high.safeCast(Py_ssize_t) + 1);
+
+      } else if (!bounds.hasLowBound() && bounds.hasHighBound()) {
+        pyObj = PyTuple_GetSlice(this.getPyObject(),
+                                 (min(int(64))).safeCast(Py_ssize_t),
+                                 bounds.high.safeCast(Py_ssize_t) + 1);
+
+      } else if (bounds.hasLowBound() && !bounds.hasHighBound()) {
+        pyObj = PyTuple_GetSlice(this.getPyObject(),
+                                 bounds.low.safeCast(Py_ssize_t),
+                                 (max(int(64))).safeCast(Py_ssize_t));
+
+      } else {
+        pyObj = PyTuple_GetSlice(this.getPyObject(),
+                                 (min(int(64))).safeCast(Py_ssize_t),
+                                 (max(int(64))).safeCast(Py_ssize_t));
```

Comment:
these look really suspect to me.

For example, `myTuple.get(1..)` is the same as `myTuple[1:]` in Python gives you everything but the first element. But here `1..` is translated to `1:2**63-1`. On a 32 bit system that will fail on the cast to Py_ssize_t, and on a 64 bit system you will just get a tuple out of bounds error.

Reply:
after reading the rest of the PR, it seems this does work. the docs make no indication it works this way. weird

My note about 32 bit platforms still stands, but that can probably be fixed by just using `max(Py_ssize_t)`